File: modulo1/programasGabarito/ex3tempoExecucao.f90
program tempo_execucao
  implicit none
  integer :: i
  real(8) :: s, t_start, t_end

  s = 0.0d0
  call cpu_time(t_start)

  do i = 1, 100000000
     s = s + 1.0d0 / dble(i)
  end do

  call cpu_time(t_end)

  print *, 'Resultado da soma: ', s
  print *, 'Tempo de execução: ', t_end - t_start, 'segundos'
end program tempo_execucao

File: modulo1/programasGabarito/ex3tempoExecucao.cpp
#include <iostream>
#include <iomanip>
#include <chrono>
using namespace std;
using namespace chrono;

int main() {
    double s = 0.0;

    auto start = high_resolution_clock::now();

    for (int i = 1; i <= 100000000; ++i)
        s += 1.0 / i;

    auto end = high_resolution_clock::now();
    duration<double> elapsed = end - start;

    cout << fixed << setprecision(6);
    cout << "Resultado da soma: " << s << endl;
    cout << "Tempo de execução: " << elapsed.count() << " segundos" << endl;

    return 0;
}

